### DISCUSSION TITLE
Confused by different interface on Autokey

### DISCUSSION THREAD

**Discussion by triplemaya** [autokey/autokey#932] [Q&A] [answered]
I am following the instructions here which are nicely presented:
http://techfulspot.blogspot.com/2015/03/remapping-keys-in-linux-using-autokey.html
But the autokey version i am using comes up with three new phrases already in the interface. Somehow when I follow the instructions I am getting one of these sent for the key, and one of my own choices for the action. Any chance the instructions could be given for the new interface arrangement.

**Comment by triplemaya** [comment]
Can't see how to edit post. Feeling an idiot! Just deleted the phrases. 
However, cannot find out how to get the name of the key to remap. It is the enter key on the number pad. 
Tried xev but it just gives me three lines of numbers.
In the windows app I used I just have to press the key I want to remap. And then in a different place press the key I want to map it to. Is there any way to do this with this linux app. (Seems like a totally sensible way to do things, but can't see how to do it. )

**Comment by josephj11** [comment, marked as answer]
Welcome to the AutoKey Community, @triplemaya !

We don't have a general tutorial like that article. It's almost nine years old and a lot has changed since then.

Just read the article. Almost all of it is still correct except for a few fine points. We keep finding these articles that we didn't know about before.

You can start by reading through our [wiki](https://github.com/autokey/autokey/wiki/). It has a bunch of helpful articles and example scripts.

Specifics:

Editing: there's an ellipsis/hamburger `...` in the upper right corner of each post on GitHub. It will open a drop down menu that will allow you to edit any post you authored.

Hotkeys: When you create/edit an action (phrase or script), there is a `Set` button to the right of `Hotkey:`. When you click on it, a pop-up appears. Click on `Record a key combination` then press the keys on the keyboard which you want to use for the hotkey.

This will work great for most things, but you might have difficulty defining a hotkey using the numeric keypad. I haven't tried it in a while.

That being said, if what you really want to do in this instance is to remap your keyboard and you want it  to work for everything and not just when certain conditions are met, then `xmodmap` or `xkb` will do that for you better than AutoKey will.

FYI:

Most distros still offer AutoKey 0.95.10 in their repos.

So, you are probably not running 0.96.0, our current release. This release contains a lot of bug fixes and a few new features. We highly encourage users to upgrade as soon as possible. You can find it [here](https://github.com/autokey/autokey/releases/) and instructions are [here](https://github.com/autokey/autokey/wiki/Installing).

The installation is pretty easy to do and we also provide .deb packages for users of Debian and Debian-derived distributions (Ubuntu, Mint, ...).

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.

New users are encouraged to spend some time browsing through our [wiki](https://github.com/autokey/autokey/wiki). There are a number of helpful articles there as well as many example AutoKey scripts.

**Comment by triplemaya** [comment]
Thank you for your long and very helpful post. 

As you say, a different program may be the obvious answer. I think I got confused because something called autokey on windows just did exactly what I wanted and expected. I vaguely thought how wonderful, a really good cross platform app. There are so few. 

However, having found the button, again I think, when I press the numpad enter it just registers as enter. Which of course is not what I want. So something more complex where I discover the identifier is probably required.

**Reply by josephj11** [reply]
AutoHotKey works on Windows. There are a couple of ports of it to Linux in various states of development. Also, Espanso claims to be able to convert AutoHotKey phrases into their system on Linux. See [this](https://github.com/autokey/autokey/discussions/775).

As for the numpad Enter being indistinguishable from normal Enter. That's a separate issue that you can file if you want to. IDK if that's an AutoKey problem or one with the environment.
